Sporadic ArithmeticException when sending WebSocket upgrade requests on 9.4.x

**Jetty version**

9.4.35.v20201120

**Java version**

1.8 (let me know if you need to know the exact update)

**OS type/version**

Linux 4.4.0-1052-aws

**Description**

We're sporadically seeing exceptions like the following when sending `WebSocketUpgradeRequests`:

```
Caused by: java.lang.ArithmeticException: / by zero
	at org.eclipse.jetty.util.Pool.acquire(Pool.java:278)
	at org.eclipse.jetty.client.AbstractConnectionPool.activate(AbstractConnectionPool.java:284)
	at org.eclipse.jetty.client.AbstractConnectionPool.acquire(AbstractConnectionPool.java:209)
	at org.eclipse.jetty.client.HttpDestination.process(HttpDestination.java:329)
	at org.eclipse.jetty.client.HttpDestination.send(HttpDestination.java:316)
	at org.eclipse.jetty.client.HttpDestination.send(HttpDestination.java:309)
	at org.eclipse.jetty.client.HttpDestination.send(HttpDestination.java:286)
	at org.eclipse.jetty.client.HttpDestination.send(HttpDestination.java:263)
	at org.eclipse.jetty.client.HttpClient.send(HttpClient.java:600)
	at org.eclipse.jetty.client.HttpRequest.sendAsync(HttpRequest.java:778)
	at org.eclipse.jetty.client.HttpRequest.send(HttpRequest.java:765)
	at org.eclipse.jetty.websocket.client.WebSocketUpgradeRequest.send(WebSocketUpgradeRequest.java:565)
	at org.eclipse.jetty.websocket.client.WebSocketUpgradeRequest.sendAsync(WebSocketUpgradeRequest.java:570)
	at com.gradle.enterprise.testdistribution.common.client.websocket.h.a(SourceFile:70)
	at com.gradle.enterprise.testdistribution.common.client.websocket.h.a(SourceFile:53)
	at com.gradle.enterprise.testdistribution.common.client.websocket.o.b(SourceFile:47)
	at net.jodah.failsafe.Functions.lambda$toCtxSupplier$15(Functions.java:276)
	at net.jodah.failsafe.Functions.lambda$getPromiseOfStage$6(Functions.java:154)
	at net.jodah.failsafe.Functions.lambda$getPromiseAsync$4(Functions.java:100)
	... 8 more
```

We've limited the `maxRequestsQueuedPerDestination` on the `HttpClient` to 4 in case that's relevant.

--- a/jetty-util/src/main/java/org/eclipse/jetty/util/Pool.java
+++ b/jetty-util/src/main/java/org/eclipse/jetty/util/Pool.java
@@ -301,6 +301,11 @@ public class Pool<T> implements AutoCloseable, Dumpable
             {
                 LOGGER.ignore(e);
                 size = entries.size();
+                // Size can be 0 when the pool is in the middle of
+                // acquiring a connection while another thread
+                // removes the last one from the pool.
+                if (size == 0)
+                    break;
             }
             index = (index + 1) % size;
         }

